### ISSUE TITLE
Feature request: propTypes shape detection

### ISSUE THREAD

**Issue by lencioni** [yannickcr/eslint-plugin-react#62]
Now that we have the [prop-types rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md), it would be nice if it could be configured to (perhaps by default) help you find properties on objects from props that you are using. For instance:

```jsx
const React = require('react/addons');

const MyComponent = React.createClass({
  propTypes: {
    myObj: React.PropTypes.object,
  },

  /**
   * @return {Object}
   */
  render: function() {
    const { myObj } = this.props;

    return (
      <div>{myObj.body}</div>
    );
  }
});

module.exports = MyComponent;
```

It would be nice if this plugin told me that `myObj.body` is missing from the shape of `myObj`. Where this would not raise any warnings/errors:

```jsx
const React = require('react/addons');

const MyComponent = React.createClass({
  propTypes: {
    myObj: React.PropTypes.shape({
      body: PropTypes.string,
    }),
  },

  /**
   * @return {Object}
   */
  render: function() {
    const { myObj } = this.props;

    return (
      <div>{myObj.body}</div>
    );
  }
});

module.exports = MyComponent;
```

**Comment by yannickcr** [comment]
Sorry, I don't think I'll do it, at least not for now.
It should be really difficult to do this properly without missing a lot of edges cases.

But if someone has a solution I can accept a PR :)
